mhlib/app_test.go

```diff
@@ -1,67 +1,68 @@
 package mhlib
 
 import (
+	"fmt"
+	"net"
 	"testing"
 )
 
-
 func TestSelfRender(t *testing.T) {
 
-	// With datasources
+	// Without datasources
 	templateString := `
-gomplate:
-  datasources:
-  - "http_obj=https://httpbin.org/get"
-  datasourceheaders: []
 values:
   a: foo
   b: '[[ .values.a ]]bar' # foobar
   c: '[[ .values.b ]]baz' # foobarbaz
-  f: 'Func Test: [[ net.LookupIP "example.com" ]]'
 `
 	expected := `
-gomplate:
-  datasources:
-  - "http_obj=https://httpbin.org/get"
-  datasourceheaders: []
 values:
   a: foo
   b: 'foobar' # foobar
   c: 'foobarbaz' # foobarbaz
-  f: 'Func Test: 10.114.236.103'
 `
-	out, err := selfRender(templateString, true)
-	if err != nil {
-		t.Log(err)
-		t.Fail()
-	}
-	if out != expected {
-		t.Logf("Actual: %s\nExpected: %s\n", out, expected)
-		t.Fail()
+	if out, err := selfRender(templateString); out != expected {
+		if err != nil {
+			t.Log(err)
+			t.FailNow()
```

Comment:
Why not run through all of the tests, and then fail later?

Reply:
This would be easier to debug a failed test.